source/logrepl/cdc.go

```diff
@@ -176,7 +181,15 @@ func (i *CDCIterator) Ack(_ context.Context, sdkPos sdk.Position) error {
 // or the context gets canceled. If the subscription stopped with an unexpected
 // error, the error is returned.
 func (i *CDCIterator) Teardown(ctx context.Context) error {
-	defer i.pgconn.Close(ctx)
+	defer func() {
+		// Give the subscription 2 seconds to settle.
+		select {
+		case <-i.sub.Done():
+		case <-time.After(subscriberDoneTimeout):
+		}
```

Comment:
This part is doing the same as `i.sub.Wait` just with an additional 2 second delay. Why don't we just change `Subscription.Wait()` to allow us to do this? For example:

```go
i.sub.Wait(ctx, subscriberDoneTimeout)
```

and

```go
func (s *Subscription) Wait(ctx context.Context, minimumWaitDelay time.Duration) error {
	select {
	case <-s.done:
		return nil
	case <-time.After(minimumWaitDelay):
		// Delay reached, continue waiting for as long as the context is open.
	}
	select {
	case <-ctx.Done():
		return ctx.Err()
	case <-s.done:
		return nil
	}
}
```